drivers/hwtrig/hwtrig_stm32_timer.c

```diff
@@ -0,0 +1,244 @@
+
+/* L0 series MCUs only have 16-bit timers and don't have below macro defined */
+#ifndef IS_TIM_32B_COUNTER_INSTANCE
+#define IS_TIM_32B_COUNTER_INSTANCE(INSTANCE) (0)
+#endif
+
+/** HWTRIG data. */
+struct hwtrig_stm32_data {
+	/** Timer clock (Hz). */
+	uint32_t tim_clk;
+};
+
+/** HWTRIG configuration. */
+struct hwtrig_stm32_config {
+	/** Timer instance. */
+	TIM_TypeDef *timer;
+	/** Prescaler. */
+	uint32_t prescaler;
+	/** Clock configuration. */
+	struct stm32_pclken pclken;
+};
+
+/**
+ * Obtain timer clock speed.
+ *
+ * @param pclken  Timer clock control subsystem.
+ * @param tim_clk Where computed timer clock will be stored.
+ *
+ * @return 0 on success, error code otherwise.
+ */
+static int get_tim_clk(const struct stm32_pclken *pclken, uint32_t *tim_clk)
+{
+	int r;
+	const struct device *clk;
+	uint32_t bus_clk, apb_psc;
+
+	clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
+
+	r = clock_control_get_rate(clk, (clock_control_subsys_t *)pclken,
+				   &bus_clk);
+	if (r < 0) {
+		return r;
+	}
+
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	if (pclken->bus == STM32_CLOCK_BUS_APB1) {
+		apb_psc = CONFIG_CLOCK_STM32_D2PPRE1;
+	} else {
+		apb_psc = CONFIG_CLOCK_STM32_D2PPRE2;
+	}
+#else
+	if (pclken->bus == STM32_CLOCK_BUS_APB1) {
+		apb_psc = CONFIG_CLOCK_STM32_APB1_PRESCALER;
+	}
+#if !defined(CONFIG_SOC_SERIES_STM32F0X) && !defined(CONFIG_SOC_SERIES_STM32G0X)
+	else {
+		apb_psc = CONFIG_CLOCK_STM32_APB2_PRESCALER;
+	}
+#endif
+#endif
+
+#if defined(RCC_DCKCFGR_TIMPRE) || defined(RCC_DCKCFGR1_TIMPRE) || \
+	defined(RCC_CFGR_TIMPRE)
+	/*
+	 * There are certain series (some F4, F7 and H7) that have the TIMPRE
+	 * bit to control the clock frequency of all the timers connected to
+	 * APB1 and APB2 domains.
+	 *
+	 * Up to a certain threshold value of APB{1,2} prescaler, timer clock
+	 * equals to HCLK. This threshold value depends on TIMPRE setting
+	 * (2 if TIMPRE=0, 4 if TIMPRE=1). Above threshold, timer clock is set
+	 * to a multiple of the APB domain clock PCLK{1,2} (2 if TIMPRE=0, 4 if
+	 * TIMPRE=1).
+	 */
+
+	if (LL_RCC_GetTIMPrescaler() == LL_RCC_TIM_PRESCALER_TWICE) {
+		/* TIMPRE = 0 */
+		if (apb_psc <= 2u) {
+			LL_RCC_ClocksTypeDef clocks;
+
+			LL_RCC_GetSystemClocksFreq(&clocks);
+			*tim_clk = clocks.HCLK_Frequency;
+		} else {
+			*tim_clk = bus_clk * 2u;
+		}
+	} else {
+		/* TIMPRE = 1 */
+		if (apb_psc <= 4u) {
+			LL_RCC_ClocksTypeDef clocks;
+
+			LL_RCC_GetSystemClocksFreq(&clocks);
+			*tim_clk = clocks.HCLK_Frequency;
+		} else {
+			*tim_clk = bus_clk * 4u;
+		}
+	}
+#else
+	/*
+	 * If the APB prescaler equals 1, the timer clock frequencies
+	 * are set to the same frequency as that of the APB domain.
+	 * Otherwise, they are set to twice (×2) the frequency of the
+	 * APB domain.
+	 */
+	if (apb_psc == 1u) {
+		*tim_clk = bus_clk;
+	} else {
+		*tim_clk = bus_clk * 2u;
+	}
+#endif
+
+	return 0;
+}
+
+static int hwtrig_stm32_enable(const struct device *dev, void *arg)
```

Comment:
Should this not be the following?
```suggestion
static int hwtrig_stm32_enable(const struct device *dev, hwtrig_flags_t arg)
```

Reply:
Thanks! I missed that. By the way, the ci was unhappy that I created `hwtrig_flags_t` type